2_process/src/munge_GFv1_catchments.R

```diff
@@ -0,0 +1,147 @@
+find_intersecting_hru <- function(prms_line,prms_hrus){
+  #' 
+  #' @description This function finds all of the HRU polygons that intersect a
+  #' PRMS segment, and returns the HRU ID (i.e., hru_segment) that has the
+  #' greatest percent of overlap with the PRMS segment
+  #' 
+  #' @param prms_line sf object containing one PRMS river segment of interest
+  #' prms_line must contain variables subsegid and geometry
+  #' @param prms_hrus sf (multi)polygon containing the HRU's from the NHGFv1
+  #
+  
+  # Transform PRMS segment and HRU's into a consistent projection
+  # using Albers Equal Area Conic CONUS
+  prms_line_proj <- sf::st_transform(prms_line, 5070)
+  prms_hrus_proj <- sf::st_transform(prms_hrus, 5070)
+  
+  # Calculate the length of prms_line and add as an attribute
+  prms_line_proj$length = sf::st_length(prms_line_proj$geometry)
```

Comment:
The `subseglen` attribute has reach length (hopefully they are similar). Are you keeping this in case a reach length is not in the attribute table? Or maybe it's a safer approach because length is also computed for the intersected segments

Reply:
Good point, I don't have a strong reason for calculating length separately. I kind of subscribed to the idea that it's safer since I also calculate the length for the intersected segments. However, I checked and for all PRMS segments the new attribute `length` was identical to `subseglen` so I've omitted this step from the code.